tf2onnx/utils.py

```diff
@@ -120,21 +120,38 @@ def split_nodename_and_shape(name):
 
 
 def tf_to_onnx_tensor(tensor, name=""):
-    """Convert tensorflow tensor to onnx tensor."""
+    """
+    Convert tensorflow tensor to onnx tensor.
+    Here deal with three types of tensor:
+    1. normal tensor, e.g., np.array([1,2,3], dtype=DTYPE):
+        tensor_content: raw data of [1,2,3]
+        tensor_shape.dim: [3]
+        DTYPE_val: empty
+    2. scalar tensor, e.g., np.array(1, dtype=DTYPE):
+        tensor_content: empty
+        tensor_shape.dim: [0]
+        DTYPE_val: 1
+    3. empty tensor, e.g., np.array([], dtype=DTYPE):
+        tensor_content: empty
+        tensor_shape.dim: [0]
+        DTYPE_val: empty
+    """
     new_type = TF_TO_ONNX_DTYPE[tensor.dtype]
     tdim = tensor.tensor_shape.dim
     dims = [d.size for d in tdim]
-    # FIXME: something is fishy here
-    if dims == [0]:
-        dims = [1]
     is_raw, data = get_tf_tensor_data(tensor)
+    # empty tensor
+    if dims == [0] and not is_raw and not data:
+        np_data = np.array([], dtype=map_onnx_to_numpy_type(new_type))
+        return numpy_helper.from_array(np_data, name=name)
```

Comment:
data should not be None from here, right?
let's add a make_sure.

Reply:
hmm, `data = None` here is just a indicator for empty tensor data.

Reply:
yep, i mean add a check AFTER dealing with empty tensor?

Reply:
oh, yes, it won't be nether None nor []